site/app/controllers/PollController.php

```diff
@@ -163,10 +163,20 @@ public function addNewPoll() {
         $poll_id = $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
         $file_path = null;
         if (isset($_FILES['image_file']) && $_FILES["image_file"]["name"] !== "") {
-            $file = $_FILES["image_file"];
-            $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $poll_id . "_" . $_FILES["image_file"]["name"]);
-            move_uploaded_file($file["tmp_name"], $file_path);
-            $this->core->getQueries()->setPollImage($poll_id, $file_path);
+            $max_size = Utils::returnBytes(ini_get('upload_max_filesize'));
+            $acceptable = ['image/jpeg','image/jpg','image/gif','image/png'];
+            if (($_FILES['image_file']['size'] >= $max_size) || ($_FILES["image_file"]["size"] == 0)) {
+                $this->core->addErrorMessage("File(s) uploaded too large.  Maximum size is " . ($max_size / 1024) . " kb.");
+            }
+            elseif ((!in_array($_FILES['image_file']['type'], $acceptable)) && (!empty($_FILES["image_file"]["type"]))) {
```

Comment:
For the file upload validation you should use 
`validateUploadedFiles` from `site/app/libraries/FileUtils.php` which will take each uploaded file and check if there are any errors like invalid name, file size, and other upload errors. 
It will return an associative array with the status of each file with the following information:

```
'name' : name of file
'type' : filetype
'error' : an array of error messages (empty if none)
'size' : size of file
'is_zip' : true if zip file
'success' : true if there were no errors
```

Once the uploaded file is validated you can use `isValidImage` also in `FileUtils.php` to check if its an acceptable image file type.

Reply:
Alright. I'm changing it use the two functions you mentioned, but I'm running into an issue with the function `validateUploadedFiles`. Is there a way to use it when we are just expecting one file to be uploaded? I can't make it work for this case, I'm getting `Warning: count(): Parameter must be an array or an object that implements Countable` that comes from the body of the function on line https://github.com/Submitty/Submitty/blob/dbee91d58f4918b5c9fbc60bae2e5989ac959f73/site/app/libraries/FileUtils.php#L549 I'm trying to call the function with the argument `FileUtils::validateUploadedFiles($_FILES['image_file']);` when the contents of `$_FILES` are: 
![image](https://user-images.githubusercontent.com/71195502/118661588-40468f00-b7bd-11eb-9c43-bcd99ff116d6.png)

Reply:
Oh maybe that function was designed only for inputs with multiple file uploads but I'll take a look a look at it later